lab_3/main.py

```diff
@@ -20,6 +21,31 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
+    if not isinstance(text, str):
+        return ()
+    patterns = ['ö', 'ü', 'ä', 'ß']
+    replacements = ['oe', 'ue', 'ae', 'ss']
+    sentences = re.split(r"[.!?] ?", text)
+    if '' in sentences:
+        sentences.remove('')
+    lst_of_sentences = [sentence.lower() for sentence in sentences]
+    new_sentence_tuple = []
+    for sentence in lst_of_sentences:
+        sentence = sentence.split(' ')
+        new_words_lst = []
+        for word in sentence:
+            for pattern, replacement in zip(patterns, replacements):
+                word.replace(pattern, replacement)
+            lst_of_letters = [letter for letter in word if letter.isalpha()]
+            if len(lst_of_letters) != 0:
+                lst_of_letters.insert(0, '_')
+                lst_of_letters.append('_')
+                new_words_lst.append(tuple(lst_of_letters))
+                print(new_words_lst)
+            if len(new_words_lst) == 0:
+                return ()
+        new_sentence_tuple.append(tuple(new_words_lst))
+    return tuple(new_sentence_tuple)
     pass
```

Comment:
Remove as it is unused.

Reply:
Do I understand correctly that it is necessary to remove "print" on line 44?

Reply:
I meant the line with `pass`.